cfgov/unprocessed/css/cf-enhancements.less

```diff
@@ -1127,6 +1127,218 @@ textarea {
+            </div>
+            <div class="input-with-btn_btn">
+                <button class="btn btn__super">Search</button>
+            </div>
+        </div>
+  tags:
+    - cf-forms
+*/
+
+.input-contains-icon {
+    position: relative;
+
+    &_before {
+        .webfont-regular;
+        position: absolute;
+        top: 0;
+        left: unit( 10px / @btn-font-size, em );
+        margin: unit( @btn-v-padding / @btn-font-size, em ) 0;
+        line-height: normal;
+        font-size: unit( 14px / @base-font-size-px, em );
+        pointer-events: none;
+    }
+
+    &.input__super {
+        .input-contains-icon_before {
+            font-size: unit( 18px / @base-font-size-px, em );
+        }
+    }
+
+    input[type="text"],
+    input[type="search"],
+    input[type="email"],
+    input[type="url"],
+    input[type="tel"],
+    input[type="number"] {
+        box-sizing: border-box;
+        width: 100%;
+    }
+
+    &_before ~ input[type="text"],
+    &_before ~ input[type="search"],
+    &_before ~ input[type="email"],
+    &_before ~ input[type="url"],
+    &_before ~ input[type="tel"],
+    &_before ~ input[type="number"] {
+        padding-left: unit( 25px / @btn-font-size, em );
+
+        &.input__super {
+            padding-left: unit( 35px / @super-btn-font-size, em );
+        }
+    }
+
+    &_after ~ input[type="text"],
+    &_after ~ input[type="search"],
+    &_after ~ input[type="email"],
+    &_after ~ input[type="url"],
+    &_after ~ input[type="tel"],
+    &_after ~ input[type="number"] {
+        padding-right: unit( 30px / @btn-font-size, em );
+
+        &.input__super {
+            padding-right: unit( 35px / @super-btn-font-size, em );
+        }
+    }
+
+    &_before .btn,
+    &_after .btn {
+        .u-link__no-border();
+        position: absolute;
+        right: unit( 15px / @btn-font-size, em );
+        top: 0;
+
+        &__super {
+            right: unit( 15px / @super-btn-font-size, em );
+        }
+
+        // pa11y fails unless the background color is explicitly set.
+        &__link.btn__secondary {
+            background-color: #fff;
+        }
+    }
+
+    // If the icon contains a text label,
+    // the label is hidden at mobile sizes to save space.
+    &_label {
+        display: none;
+
+        .respond-to-min( @bp-sm-min, {
+            display: inline-block;
+        } );
+    }
+
+    // Specific adjustments for particular icon/label combinations.
+
+    // Add magnifying glass icon to input field.
+    &_before__search {
+        .input-contains-icon_icon:before {
+            .cf-icon();
+            content: @cf-icon-search;
+        }
+    }
+
+    // Add "x Clear" button to input field.
+    &_after__clear {
+        .input-contains-icon_icon:before {
+            .cf-icon();
+            content: @cf-icon-delete;
```

Comment:
Has your team discussed the possibility of using the delete icon with the circular background, to avoid any possible confusion with the text of the input?